assets/js/base/context/cart-checkout/checkout-state/index.js

```diff
@@ -160,8 +160,11 @@ export const CheckoutStateProvider = ( {
 				}
 			} );
 		}
-		if ( status === STATUS.COMPLETE ) {
-			if ( hasError ) {
+	}, [ checkoutState.status, setValidationErrors ] );
+
+	useEffect( () => {
+		if ( checkoutState.status === STATUS.COMPLETE ) {
+			if ( checkoutState.hasError ) {
```

Comment:
I split up the event emitters into separate effects because in my testing it seemed to reduce the amount of times the effects were called.